src/test/java/de/tum/in/www1/artemis/domain/ShortAnswerSubmittedTextTest.java

```diff
@@ -0,0 +1,114 @@
+    public void init() {
+        shortAnswerQuestion = new ShortAnswerQuestion();
+        ShortAnswerSubmittedAnswer shortAnswerSubmittedAnswer = new ShortAnswerSubmittedAnswer();
+        shortAnswerSubmittedAnswer.setQuizQuestion(shortAnswerQuestion);
+        shortAnswerSubmittedText = new ShortAnswerSubmittedText();
+        shortAnswerSubmittedText.setSubmittedAnswer(shortAnswerSubmittedAnswer);
+    }
+
+    /**
+     * @func testSubmissionWithMatchingLetterCaseAndExactMatching
+     * @desc tests {@link ShortAnswerSubmittedText#isSubmittedTextCorrect(String, String)} with similarity value 100 and letter case matching
+     */
+    @Test
+    public void testSubmissionWithMatchingLetterCaseAndExactMatching() {
+        shortAnswerQuestion.setMatchLetterCase(true);
+        shortAnswerQuestion.setSimilarityValue(100);
+
+        String solution = "Example solution";
+
+        boolean observed = shortAnswerSubmittedText.isSubmittedTextCorrect("Example solution", solution);
+
+        assertTrue(observed);
+
+        solution = "Exampl slution";
+
+        observed = shortAnswerSubmittedText.isSubmittedTextCorrect("Example solution", solution);
+
+        assertFalse(observed);
+
+        solution = "example solution";
+
+        observed = shortAnswerSubmittedText.isSubmittedTextCorrect("Example solution", solution);
+
+        assertFalse(observed);
+    }
+
+    /**
+     * @func testSubmissionWithoutMatchingLetterCaseAndExactMatching
+     * @desc tests {@link ShortAnswerSubmittedText#isSubmittedTextCorrect(String, String)} with similarity value 100 and without letter case matching
+     */
+    @Test
+    public void testSubmissionWithoutMatchingLetterCaseAndExactMatching() {
+        shortAnswerQuestion.setMatchLetterCase(false);
+        shortAnswerQuestion.setSimilarityValue(100);
+
+        String solution = "Example solution";
+
+        boolean observed = shortAnswerSubmittedText.isSubmittedTextCorrect("Example solution", solution);
+
+        assertTrue(observed);
+
+        solution = "Exampl slution";
+
+        observed = shortAnswerSubmittedText.isSubmittedTextCorrect("Example solution", solution);
+
+        assertFalse(observed);
+
+        solution = "example solution";
+
+        observed = shortAnswerSubmittedText.isSubmittedTextCorrect("Example solution", solution);
+
+        assertTrue(observed);
+    }
+
+    /**
+     * @func testSubmissionWithNonExactMatching
+     * @desc tests {@link ShortAnswerSubmittedText#isSubmittedTextCorrect(String, String)} with similarity value 70 and without letter case matching
+     */
+    @Test
+    public void testSubmissionWithNonExactMatching() {
+        shortAnswerQuestion.setMatchLetterCase(false);
+        shortAnswerQuestion.setSimilarityValue(70);
+
+        String solution = "Example solution";
+
+        boolean observed = shortAnswerSubmittedText.isSubmittedTextCorrect("Example solut", solution);
+
+        assertTrue(observed);
+
+        solution = "Exampl slution";
```

Comment:
Example

Reply:
The idea here is to test submissions with typos and missing letters

Reply:
Ah ok!